src/test/kotlin/org/rust/ide/annotator/fixes/AddTurbofishFixTest.kt

```diff
@@ -48,14 +47,54 @@ class AddTurbofishFixTest : RsAnnotatorTestBase() {
         }
         """)
 
-    fun `test add turbofish should not be available if the instance is corrected yet`() = checkNoIntention(intention,
+    fun `test should recognize also generics arguments`() {
+        for (gen in listOf("<Option<String>>",
+            "<Option<Option<String>>>",
+            "<Option<Option<Option<String>>>>",
+            "<Option<Option<Option<Option<String>>>>>")) {
+            checkQuickFix(
+                """
+                    fn foo() {
+                        let _ = something$gen(/*caret*/"42");
+                    }
+                """, """
+                    fn foo() {
+                        let _ = something::$gen(/*caret*/"42");
+                    }
+                """)
+        }
+    }
+
+    fun `test should guess the correct boundary`() {
+        val cases = listOf(
+            Pair("", " >> 3"), Pair("", " > 3 > 5"), Pair("", " == call_something()"),
+            Pair("3 < ", ""), Pair("3 == ", ""), Pair("bye() == 3 < ", " > more() >> 2")
+        )
+
+        for ((before, after) in cases) {
```

Comment:
Ok sounds good. Just a note: by use `before` and `after` the test intent become clear but find then in string make things too complicated...

I'll change it